### ISSUE TITLE
Z3Prover JavaAPI missing interface “getConsequence”

### ISSUE THREAD

**Issue by MaxDing** [Z3Prover/z3#3202]
To retrieve fixed assignments to the set of variables in the form of consequences. I can use consequence interface of Solver class in Z3Py. It's been written in the Z3Py API docs.
```
Determine fixed values for the variables based on the solver state and assumptions.
>>> s = Solver()
>>> a, b, c, d = Bools('a b c d')
>>> s.add(Implies(a,b), Implies(b, c))
>>> s.consequences([a],[b,c,d])
(sat, [Implies(a, b), Implies(a, c)])
>>> s.consequences([Not(c),d],[a,b,c,d])
(sat, [Implies(d, d), Implies(Not(c), Not(c)), Implies(Not(c), Not(b)), Implies(Not(c), Not(a))])
```
And I can find a similar API in C/C++/.Net. But I can't find the same in JavaAPI. Is this a missing in z3 JavaAPI or something else?

PS: My z3 version is z3-4.8.7-x64-ubuntu-16.04
StackOverflow link:https://stackoverflow.com/questions/60576215/z3prover-javaapi-missing-getconsequence

**Comment by MaxDing** [comment]
@NikolajBjorner Sorry for bothering. When i was testing the `getConsequence` API with the latest code. It gave me an Exception
   `Exception in thread "main" java.lang.ClassCastException: com.microsoft.z3.AST cannot be cast to com.microsoft.z3.BoolExpr`.

I think it's the error when you try to cast `AST` to `BoolExpr` in `consequences.add((BoolExpr)result.get(i));`(Solver.java line 267). So I changed the code as below:

   ```
   public Status getConsequences(BoolExpr[] assumptions, Expr[] variables, List<BoolExpr> consequences) {
        ASTVector result = new ASTVector(getContext());
        ASTVector asms = new ASTVector(getContext());
        ASTVector vars = new ASTVector(getContext());
        for (BoolExpr asm : assumptions)
            asms.push(asm);
        for (Expr v : variables)
            vars.push(v);
        int r = Native.solverGetConsequences(getContext().nCtx(), getNativeObject(), asms.getNativeObject(),
                vars.getNativeObject(), result.getNativeObject());
        BoolExpr[] resultArray = result.ToBoolExprArray();
        for (int i = 0; i < resultArray.length; ++i) {
            consequences.add(resultArray[i]);
        }
        return lboolToStatus(Z3_lbool.fromInt(r));
    }
   ```

Then it works fine.  Am i wrong or it's actually a bug in the latest code?

**Comment by dbinkele** [comment]
This fix seems not to be part of 4.8.7 (it was released 19.11.19). Are there any plans to have an official release with this fix in a timely manner?
